src/cldr.js

```diff
@@ -107,11 +107,14 @@ export function createDateTimeFormat(format) {
     formatObj.pattern = formatObj.pattern.replace(/'([^']*)'/g, function ($0, literal) {
         return literal ? literal : "'";
     });
+    // Pattern for 12 hours inherit from pattern by default.
+    formatObj.pattern12 = formatObj.pattern;
```

Comment:
guarantee that pattern12 is defined always, that way we can be confidence that no matter what the algo is doing, it can select `pattern` or `pattern12` and something will be displayed.